### PR TITLE
Migrate recipe to convert getStatus to getStatusCode with correct types

### PR DESCRIPTION
**Pull request by mslowiak**
## What's changed?
Address migration of `getStatus` to `getStatusCode` with respecting type change from `HttpStatus` to `HttpStatusCode`. 

## What's your motivation?
- https://github.com/openrewrite/rewrite-spring/issues/554

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/spring-framework-60.yml [hunk 1]
```diff
@@ -39,9 +39,10 @@ displayName: Migrate breaking changes in `ResponseStatusException`
 description: Migrate Spring Framework 5.3's `ResponseStatusException` method `getRawStatusCode()` to Spring Framework 6's `getStatusCode().value()` and `ResponseStatusException` method `getStatus()` to Spring Framework 6's `getStatusCode()` .
 recipeList:
   - org.openrewrite.java.spring.framework.MigrateResponseStatusExceptionGetRawStatusCodeMethod
-  - org.openrewrite.java.ChangeMethodName:
-        methodPattern: "org.springframework.web.server.ResponseStatusException getStatus()"
-        newMethodName: "getStatusCode"
+  - org.openrewrite.java.spring.framework.MigrateResponseStatusExceptionGetStatusCodeMethod
+#  - org.openrewrite.java.ChangeMethodName:
+#        methodPattern: "org.springframework.web.server.ResponseStatusException getStatus()"
+#        newMethodName: "getStatusCode"
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/spring-framework-60.yml:45]:
```suggestion
```

File: src/main/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionGetStatusCodeMethod.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        .build()
+                        .apply(getCursor(), method.getCoordinates().replace(), method.getSelect());
+            }
+            return super.visitMethodInvocation(method, executionContext);
+        }
+
+        @Override
+        public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
+            if (multiVariable.getTypeExpression() != null && TypeUtils.isOfClassType(multiVariable.getTypeExpression().getType(), HTTP_STATUS_CLASS)) {
+                multiVariable = multiVariable.withTypeExpression(
```
Comment (Laurens-W) [inline src/main/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionGetStatusCodeMethod.java:64]: The variableType of the variables may still reference `HttpStatus` after this

File: src/main/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionGetStatusCodeMethod.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    static class CustomVisitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            if (method.getMethodType() != null && method.getMethodType().getDeclaringType().getFullyQualifiedName().equals("org.springframework.web.server.ResponseStatusException") &&
+                    method.getSimpleName().equals("getStatus")) {
+                return JavaTemplate.builder("#{any()}.getStatusCode()")
+                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "spring-core-6", "spring-beans-6", "spring-web-6"))
+                        .build()
+                        .apply(getCursor(), method.getCoordinates().replace(), method.getSelect());
```
Comment (Laurens-W) [inline src/main/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionGetStatusCodeMethod.java:56]: The `type` of the `fieldType` of `method.getSelect` still references `HttpStatus` however I'm not sure if that's blocking the removal of the import

File: src/test/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionTest.java [hunk 4]
```diff
@@ -43,18 +43,20 @@ void migrateResponseStatusExceptionGetStatusMethod() {
...
               """,
             """
-              import org.springframework.http.HttpStatus;
+              import org.springframework.http.HttpStatusCode;
               import org.springframework.web.server.ResponseStatusException;
+              
               class A {
                   void foo(ResponseStatusException e) {
-                      HttpStatus i = e.getStatusCode();
+                      HttpStatusCode i = e.getStatusCode();
```
Comment (Laurens-W) [inline src/test/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionTest.java:59]: identifier `i` is still of type `HttpStatus` after the recipe run

**Comment by mslowiak** [comment]
@timtebeek @Laurens-W 
Thanks for the tips, indeed it was a problem in `visitVariableDeclarations` - the `variable` in `variableDeclarations.variables[0]` had still `HttpStatus` inside `variableType`, `name.type`, `name.fieldType` 🚀

**Review by timtebeek — APPROVED** [review]
Great to see you've worked it out; Thanks a lot! Should help smooth out upgrade where the return type changed.

**Pull request merged**